name: reproduce-probe-failure-issue
apiVersion: automation.cloudbees.io/v1alpha1
kind: workflow

on:
  workflow_dispatch: {}

jobs:
  test-probe-failure:
    permissions:
      scm-token-own: read
      id-token: write
    steps:
    - uses: cloudbees-io/checkout@v1
    
    - uses: https://github.com/cloudbees-io/configure-aws-credentials@v1
      with:
        aws-region: us-east-1
        role-to-assume: ${{ vars.oidc_staging_iam_role }}
        role-duration-seconds: "3600"
    
    - uses: cloudbees-io/configure-ecr-credentials@v1
    
    - uses: cloudbees-io/configure-eks-credentials@v1
      with:
        name: ${{ vars.staging_east_cluster_name }}
    
    - id: createns
      uses: cloudbees-io/create-k8s-namespace@v1
      with:
        name: ${{ format('probe-test-{0}', cloudbees.scm.sha) }}
        sanitize-name: "true"
        labels: "cloudbees.io/cleanup: true"
    
    # This step reproduces the exact issue from the ticket
    - name: Deploy chart with failing probes (reproduces the issue)
      uses: cloudbees-io/helm-install@v1
      with:
        release-name: failing-probe-test
        chart-location: charts/failing-probe-chart
        namespace: ${{ steps.createns.outputs.name }}
        timeout: 2m
        wait: 'true'